File: 1013-fibonacci-number/fibonacci-number.py
class Solution(object):
    def fib(self, n):
        """
        :type n: int
        :rtype: int
        """
        nums = [0, 1]
        while len(nums) <= n:
            nums.append(nums[len(nums) - 1] + nums[len(nums) - 2])
        return nums[n]

File: 1013-fibonacci-number/fibonacci-number.cpp
class Solution {
public:
    int fib(int n) {
        if (!n) return 0;
        
        int first = 0;
        int second = 1;
        for (int i = 1; i < n; ++i){
            int temp = first + second;
            first = second;
            second = temp;
        }

        return second;
    }
};